pingcap/ticdc
# processor: prevents individual table pipelines to block the entire processor

### What problem does this PR solve? 
- In the current implementation of the new processor, if the pipeline for any table is blocked, the entire processor will be blocked.

### What is changed and how it works?
- Skip sending messages to the blocked pipeline. Note that this only means stop sending control messages to the table pipeline, and any back-pressure handling mechanism for the data critical path remains unchanged.

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test

Start 10 tpcc changefeeds, and wait for incremental scans to finish.
`curl "http://xxx:8300/debug/fail/github.com/pingcap/ticdc/pkg/pipeline/PipelineSendToFirstNodeTryAgain" -X PUT -d '40%return(true)'`
Wait for 1 minute and review log to see that the failpoint has taken effect.
`curl "http://xxx:8300/debug/fail/github.com/pingcap/ticdc/pkg/pipeline/PipelineSendToFirstNodeTryAgain" -X PUT -d 'return(true)'`
Wait for 1 minute to observe that checkpoint-ts has stopped updating.
`curl "http://xxx:8300/debug/fail/github.com/pingcap/ticdc/pkg/pipeline/PipelineSendToFirstNodeTryAgain" -X DELETE`
Wait until checkpoint-ts resumes updating.

Related changes

 - Need to cherry-pick to the release branch


### Release note

- No release note
